kaiax/gasless/README.md

```diff
@@ -1,32 +1,70 @@
+Note that gasless transaction does not mean the gas prices of gasless transactions are zero, but it means that proposer will lend the user with gas fee and user will pay back during gasless swap.
+
+The gas fee of gasless transaction's is funded by block proposer (i.e., lend transaction generated by `GetLendTxGenerator`), and user repays the lent amount during gasless swap.
 
 ### Transaction pool rules
 
-TBU
+#### Ready
+
+This module is responsible for promoting gasless transactions.
+Sender's nonce of GaslessSwapTx is checked to distinguish if GaslessApproveTx is expected. If `tx.nonce == GetNonce(sender) + 1`, GaslessApproveTx is expected. If `tx.nonce == GetNonce(sender)`, GaslessApproveTx is not expected.
+
+If GaslessApproveTx is expected, GaslessApproveTx and GaslessSwapTx can be promoted when they are both ready for execution.
+Otherwise, GaslessSwapTx can be promoted when it is ready for execution.
+
+See ready condition [KIP-247](https://kips.kaia.io/KIPs/kip-247) and the implementation `IsExecutable(approveTxOrNil, swapTx *types.Transaction) bool`.
+
+#### Balance check
+
+Sender balance check is omitted for gasless transactions (see `GetCheckBalance()`).
 
 ### Block building rules
 
-TBU
+Upon detection of GaslessTxs, the following logics are executed:
+
+- Per sender, if exists, GaslessApproveTx is relocated before GaslessSwapTx.
+- LendTxGenerator is prepended before GaslessApproveTx.
+- A new bundle is generated which contain either `[LendTxGenerator, GaslessApproveTx, GaslessSwapTx]` or `[LendTxGenerator, GaslessSwapTx]`
+- If the bundle has conflict with any previous bundles, it is excluded from the returned bundle list.
+
+## Persistent schema
 
-## In-memory structures
+This module does not persist any data.
 
 ## Module lifecycle
 
+### Init
+
+- Dependencies:
+  - ChainConfig: to generate the latest signer.
+  - NodeKey: for LendTxGenerator.
+  - TxPool: to fetch latest nonce of sender.
+- Notable dependents:
+  - worker: to extract bundles.
+
+### Start and stop
+
+This module does not have any background threads.
+
+## Block processing
+
+### Execution
+
+This module reads every block `GaslessSwapRouter` contract to detect any changes in allowed tokens.
+
 ## APIs
 
 ## Getters
 
-- IsGaslessApproveTx(tx *types.Transaction) bool
-- IsGaslessSwapTx(tx *types.Transaction) bool
-- IsGaslessPattern(approveTxOrNil, swapTx *types.Transaction) bool
-- MakeLendTx(approveTxOrNil, swapTx *types.Transaction) (*types.Transaction, error)
+- `IsApproveTx(tx *types.Transaction) bool`: Returns if given transaction satisfies all conditions to be recognized as GaslessApproveTx.
+- `IsSwapTx(tx *types.Transaction) bool`: Returns if given transaction satisfies all conditions to be recognized as GaslessSwapTx.
+- `IsExecutable(approveTxOrNil, swapTx *types.Transaction) bool`: Returns if gasless transactions are ready to be executed.
+- `GetLendTxGenerator(approveTxOrNil, swapTx *types.Transaction) *builder.TxOrGen`: Returns LendTxGenerator.
```

Comment:
Certainly, these are getters functions but these are used for only the module internally. How about providing the [GaslessModule interface](https://github.com/kaiachain/kaia/blob/af3023d0de7aba7a31312800c1831117de98e7d9/kaiax/gasless/interface.go#L25-L30)?

Reply:
Only module-specific getters are described, as in

https://github.com/kaiachain/kaia/blob/6426d4be8b09cfd16c89a893bdeac08f94f4589d/kaiax/staking/interface.go#L26-L34
and
https://github.com/kaiachain/kaia/blob/6426d4be8b09cfd16c89a893bdeac08f94f4589d/kaiax/staking/README.md#L208-L213

GaslessModule interface is generic kaiax Modules, so I think it's ok without their descriptions.
For consistency with other modules' READMEs, I'll just remove the getters.